src/StaticContent/ThemeResolver.php

```diff
@@ -0,0 +1,125 @@
+use Magento\MagentoCloud\Filesystem\FileSystemException;
+use Magento\MagentoCloud\Package\UndefinedPackageException;
+use Psr\Log\LoggerInterface;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+
+/**
+ * Resolves themes to their correct names
+ */
+class ThemeResolver
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var File
+     */
+    private $file;
+
+    /**
+     * @var DirectoryList
+     */
+    private $directoryList;
+
+    /**
+     * @var string[]
+     */
+    private $themes;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param File $file
+     * @param DirectoryList $directoryList
+     */
+    public function __construct(LoggerInterface $logger, File $file, DirectoryList $directoryList)
+    {
+        $this->logger = $logger;
+        $this->file = $file;
+        $this->directoryList = $directoryList;
+    }
+
+    /**
+     * Takes in name of a theme, compares it against the names and corrects if necessary.
+     *
+     * @param string $themeName
+     * @return string
+     * @throws UndefinedPackageException
+     */
+    public function resolve(string $themeName): string
+    {
+        $availableThemes = $this->getThemes();
+        if (!in_array($themeName, $availableThemes)) {
+            $this->logger->warning('Theme ' . $themeName . ' does not exist.');
+            $themeNamePosition = array_search(
+                strtolower($themeName),
+                array_map('strtolower', $availableThemes)
+            );
+            if (false !== $themeNamePosition) {
+                $this->logger->warning(
+                    'Theme found as ' . $availableThemes[$themeNamePosition] . '.  Using corrected name instead.'
+                );
+                return $availableThemes[$themeNamePosition];
+            }
+        }
+        return '';
+    }
+
+    /**
+     * @return array
+     * @throws UndefinedPackageException
+     */
+    public function getThemes(): array
+    {
+        if (empty($this->themes)) {
+            $this->logger->debug('Finding themes from disk.');
+            $this->themes = array_merge(
+                $this->file->glob($this->directoryList->getPath(DirectoryList::DIR_DESIGN, true) . '/*/*/*/theme.xml'),
+                $this->file->glob($this->directoryList->getPath(DirectoryList::DIR_VENDOR, true) . '/*/*/theme.xml')
+            );
+            foreach ($this->themes as &$themePath) {
+                $themePath = $this->getThemeName(substr($themePath, 0, strrpos($themePath, 'theme.xml')));
+            }
+            $this->logger->debug('End of finding themes from disk.');
+        }
+
+        return $this->themes;
+    }
+
+    /**
+     * Parses the registration file of the theme to get the name of the theme as registered.
+     * @param string $themePath
+     * @return string
+     */
+    public function getThemeName(string $themePath): string
+    {
+        try {
```

Comment:
I see you already adding coupling on Magento. So maybe it's better to retrieve them from `\Magento\Framework\Component\ComponentRegistrar:$paths`?

I see it's private but we can use reflection easily:

```
if (class_exists(\Magento\Framework\Component\ComponentRegistrar::class)) {
    $reflectionClass = new \ReflectionClass(\Magento\Framework\Component\ComponentRegistrar::class);
    $property = $reflectionClass->getProperty('paths');
    $property->setAccessible(true);

    $paths = $property->getValue($reflectionClass)['theme'];

    die(var_dump(array_keys($paths)));
}
```